contracts/service-provider-directory/src/contract.rs

```diff
@@ -0,0 +1,330 @@
+    use super::*;
+    use crate::{msg::ServiceInfo, state::ServiceId};
+
+    pub fn query_id(deps: Deps, _env: Env, service_id: ServiceId) -> StdResult<ServiceInfo> {
+        let service = SERVICES.load(deps.storage, service_id)?;
+        Ok(ServiceInfo {
+            service_id,
+            service,
+        })
+    }
+
+    pub fn query_all(deps: Deps) -> StdResult<ServicesListResponse> {
+        let services = SERVICES
+            .range(deps.storage, None, None, Order::Ascending)
+            .map(|item| {
+                item.map(|(service_id, service)| ServiceInfo {
+                    service_id,
+                    service,
+                })
+            })
+            .collect::<StdResult<Vec<_>>>()?;
+        Ok(ServicesListResponse { services })
+    }
+
+    pub fn query_config(deps: Deps) -> StdResult<ConfigResponse> {
+        let config = CONFIG.load(deps.storage)?;
+        Ok(config.into())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use crate::{
+        msg::ServiceInfo,
+        test_helpers::{
+            assert::{
+                assert_config, assert_empty, assert_not_found, assert_service, assert_services,
+            },
+            fixture::service_fixture,
+            helpers::get_attribute,
+        },
+    };
+
+    use cosmwasm_std::{
+        testing::{mock_dependencies, mock_env, mock_info},
+        Addr, Coin,
+    };
+
+    #[test]
+    fn instantiate_contract() {
+        let mut deps = mock_dependencies();
+
+        let updater_role = Addr::unchecked("foo");
+        let admin = Addr::unchecked("bar");
+        let msg = InstantiateMsg {
+            updater_role: updater_role.clone(),
+            admin: admin.clone(),
+            deposit_required: Coin::new(100, "unym"),
+        };
+        let info = mock_info("creator", &[]);
+
+        // Instantiate contract
+        let res = instantiate(deps.as_mut(), mock_env(), info, msg).unwrap();
+        assert_eq!(res.messages.len(), 0);
+
+        // Check that it worked by querying the config, and checking that the list of services is
+        // empty
+        assert_config(deps.as_ref(), updater_role, admin);
+        assert_empty(deps.as_ref());
+    }
+
+    #[test]
+    fn announce() {
+        let mut deps = mock_dependencies();
+
+        let updater_role = Addr::unchecked("foo");
+        let admin = Addr::unchecked("bar");
+        let msg = InstantiateMsg {
+            updater_role: updater_role.clone(),
+            admin: admin.clone(),
+            deposit_required: Coin::new(100, "unym"),
+        };
+        let info = mock_info("creator", &[]);
+
+        // Instantiate contract
+        let res = instantiate(deps.as_mut(), mock_env(), info, msg).unwrap();
+        assert_eq!(res.messages.len(), 0);
+
+        // Announce
+        let msg = service_fixture().into_announce_msg();
+        let info = mock_info("anyone", &[]);
+        let res = execute(deps.as_mut(), mock_env(), info, msg).unwrap();
```

Comment:
but would you have expected it to work? You didn't send any funds with your request